packages/core/contracts/oracle/implementation/SkinnyOptimisticOracle.sol

```diff
@@ -0,0 +1,658 @@
+
+    // Finder to provide addresses for DVM contracts.
+    FinderInterface public finder;
+
+    // Default liveness value for all price requests.
+    uint256 public defaultLiveness;
+
+    /**
+     * @notice Constructor.
+     * @param _liveness default liveness applied to each price request.
+     * @param _finderAddress finder to use to get addresses of DVM contracts.
+     * @param _timerAddress address of the timer contract. Should be 0x0 in prod.
+     */
+    constructor(
+        uint256 _liveness,
+        address _finderAddress,
+        address _timerAddress
+    ) Testable(_timerAddress) {
+        finder = FinderInterface(_finderAddress);
+        _validateLiveness(_liveness);
+        defaultLiveness = _liveness;
+    }
+
+    /**
+     * @notice Requests a new price.
+     * @param _identifier price identifier being requested.
+     * @param _timestamp timestamp of the price being requested.
+     * @param _ancillaryData ancillary data representing additional args being passed with the price request.
+     * @param _currency ERC20 token used for payment of rewards and fees. Must be approved for use with the DVM.
+     * @param _reward reward offered to a successful proposer. Will be pulled from the caller. Note: this can be 0,
+     *               which could make sense if the contract requests and proposes the value in the same call or
+     *               provides its own reward system.
+     * @param _bond custom proposal bond to set for request. If set to 0, defaults to the final fee.
+     * @param _customLiveness custom proposal liveness to set for request.
+     * @return totalBond default bond + final fee that the proposer and disputer will be required to pay.
+     */
+    function requestPrice(
+        bytes32 _identifier,
+        uint256 _timestamp,
+        bytes memory _ancillaryData,
+        IERC20 _currency,
+        uint256 _reward,
+        uint256 _bond,
+        uint256 _customLiveness
+    ) external override nonReentrant() returns (uint256 totalBond) {
+        bytes32 requestId = _getId(msg.sender, _identifier, _timestamp, _ancillaryData);
+        require(requests[requestId] == bytes32(0), "Request already initialized");
+        require(_getIdentifierWhitelist().isIdentifierSupported(_identifier), "Unsupported identifier");
+        require(_getCollateralWhitelist().isOnWhitelist(address(_currency)), "Unsupported currency");
+        require(_timestamp <= getCurrentTime(), "Timestamp in future");
+        require(
+            _stampAncillaryData(_ancillaryData, msg.sender).length <= ancillaryBytesLimit,
+            "Ancillary Data too long"
+        );
+        uint256 finalFee = _getStore().computeFinalFee(address(_currency)).rawValue;
+
+        // Associate new request with ID
+        Request memory request;
+        request.currency = _currency;
+        request.reward = _reward;
+        request.finalFee = finalFee;
+        request.bond = _bond != 0 ? _bond : finalFee;
```

Comment:
Do you think we could instead just use whatever bond the user provides, even if it's less than the final fee?

Reply:
I changed this line to just set `request.bond = _bond` which I believe is consistent with original OO

Reply:
Sorta, except that it doesn't allow a default. Basically, it can't match the original OO because there was always the option of _not_ customizing the bond. Thinking about this more, I liked the way you were originally doing it. I think a default is preferable to just allowing 0.

Reply:
I think allowing 0 is fine to be honest, I'd rather give caller more flexibility. Setting a 0 bond means that the request is more likely to get proposed an incorrect price @chrismaree wdyt

Reply:
I think 0 could be problematic as it means there is no cost with bad proposals. however, how do you choose a reasonable default for all currencies, especially given different decimals for different erc20s?

Reply:
But with the OO you can set `bond` to be 0 via `setCustomLiveness`. The only difference here is that there is not default. The old default was the final fee for the erc20 so choosing for each token is easy.

Reply:
@nicholaspai I'm a fan of your original implementation `request.bond = _bond != 0 ? _bond : finalFee;`. If people really want 0, they can do 1 wei (they have to pay the final fee anyway, so it won't be like the proposer is actually paying nothing).